client/emojis.lua

```diff
@@ -119,64 +116,98 @@ local function DrawText3D(coords, text)
     ClearDrawOrigin()
 end
 
--- Function to display emoji above the ped's head
-local function DisplayEmoji(ped, emoji)
-    local playerPed = PlayerPedId()
-    local playerCoords = GetEntityCoords(playerPed)
-    local pedCoords = GetEntityCoords(ped)
-    local dist = #(playerCoords - pedCoords)
-
-    if dist <= 25 then
-        emojiDisplaying[ped] = (emojiDisplaying[ped] or 0) + 1
-        local isDisplaying = true
-
-        CreateThread(function()
-            Wait(5000) -- Display time in milliseconds
-            isDisplaying = false
-        end)
-
-        local offset = 1.0 + emojiDisplaying[ped] * 0.1
-        while isDisplaying do
-            if HasEntityClearLosToEntity(playerPed, ped, 17) then
-                local x, y, z = table.unpack(GetEntityCoords(ped))
-                z = z + offset
-                DrawText3D(vector3(x, y, z), emoji)
+local function startEmojiDrawLoop()
+    if drawLoopActive then return end
+    drawLoopActive = true
+
+    CreateThread(function()
+        local playerPed = PlayerPedId()
+
+        while drawLoopActive do
+            playerPed = PlayerPedId()
+            local playerCoords = GetEntityCoords(playerPed)
+            local hasActiveEmojis = false
+
+            for ped, emojiList in pairs(activeEmojis) do
+                if DoesEntityExist(ped) then
+                    local pedCoords = GetEntityCoords(ped)
+                    local dist = #(playerCoords - pedCoords)
+
+                    if dist <= 25 then
+                        for i = #emojiList, 1, -1 do
+                            local emojiData = emojiList[i]
+
+                            if GetGameTimer() >= emojiData.expireTime then
+                                table.remove(emojiList, i)
+                            else
+                                hasActiveEmojis = true
+
+                                if HasEntityClearLosToEntity(playerPed, ped, 17) then
+                                    local coords = GetEntityCoords(ped)
+                                    local offset = 1.0 + (i - 1) * 0.15
+                                    local drawCoords = vector3(coords.x, coords.y, coords.z + offset)
+                                    drawText3D(drawCoords, emojiData.emoji)
+                                end
+                            end
+                        end
+
+                        if #emojiList == 0 then
+                            activeEmojis[ped] = nil
+                        end
+                    else
+                        activeEmojis[ped] = nil
+                    end
+                else
+                    activeEmojis[ped] = nil
+                end
+            end
+
+            if not hasActiveEmojis then
```

Comment:
Get rid of the hasActiveEmojis variable and just check if table.next(activeEmojis[ped]) to determine the same thing.